src/pgduckdb_types.cpp

```diff
@@ -702,6 +702,153 @@ ConvertDecimal(const NumericVar &numeric) {
 	return (NumericIsNegative(numeric) ? -base_res : base_res);
 }
 
+duckdb::Value
+ConvertPostgresToDuckValue(Datum value, Oid postgres_type) {
+	switch (postgres_type) {
+	case BOOLOID:
+		return duckdb::Value::BOOLEAN(DatumGetBool(value));
+	case INT2OID:
+		return duckdb::Value::SMALLINT(DatumGetInt16(value));
+	case INT4OID:
+		return duckdb::Value::INTEGER(DatumGetInt32(value));
+	case INT8OID:
+		return duckdb::Value::BIGINT(DatumGetInt64(value));
+	case BPCHAROID:
+	case TEXTOID:
+	// TODO: Is this the correct place for JSONOID? I'm unable to use it in a
+	// test so I don't know.
+	// case JSONOID:
+	case VARCHAROID: {
+		return duckdb::Value((const char *)value);
```

Comment:
Don't we need the whole de-toastification logic here?

Reply:
Toast is only used when storing values on disk. Since this is a parameter passed in by the client we don't need it here (assuming we only ever pass parameter values to the function). I added this consideration to the function comment now.